x/pool-incentives/keeper/distr_test.go

```diff
@@ -56,280 +56,299 @@ func (suite *KeeperTestSuite) TestAllocateAssetToCommunityPoolWhenNoDistrRecords
-		GaugeId: 0,
-		Weight:  sdk.NewInt(100),
-	}, types.DistrRecord{
-		GaugeId: gaugeId,
-		Weight:  sdk.NewInt(100),
-	})
-	suite.NoError(err)
-	distrInfo := keeper.GetDistrInfo(suite.Ctx)
-	suite.Equal(2, len(distrInfo.Records))
-	suite.Equal(sdk.NewInt(100), distrInfo.Records[0].Weight)
-	suite.Equal(sdk.NewInt(100), distrInfo.Records[1].Weight)
-	suite.Equal(sdk.NewInt(200), distrInfo.TotalWeight)
-
-	// adding two of the same gauge id at once should error
-	err = keeper.UpdateDistrRecords(suite.Ctx, types.DistrRecord{
-		GaugeId: 0,
-		Weight:  sdk.NewInt(0),
-	}, types.DistrRecord{
-		GaugeId: gaugeId,
-		Weight:  sdk.NewInt(200),
-	}, types.DistrRecord{
-		GaugeId: gaugeId + 1,
-		Weight:  sdk.NewInt(150),
-	})
-	suite.NoError(err)
-	distrInfo = keeper.GetDistrInfo(suite.Ctx)
-	suite.Equal(2, len(distrInfo.Records))
-	suite.Equal(sdk.NewInt(200), distrInfo.Records[0].Weight)
-	suite.Equal(sdk.NewInt(150), distrInfo.Records[1].Weight)
-	suite.Equal(sdk.NewInt(350), distrInfo.TotalWeight, distrInfo.TotalWeight.String())
+func (suite *KeeperTestSuite) TestUpdateDistrRecords() {
+	tests := []struct {
+		name           string
+		args           []types.DistrRecord
+		isPoolPrepared bool
+		expectErr      bool
+	}{
+		{
+			name: "Not existent gauge.",
+			args: []types.DistrRecord{{
+				GaugeId: 1,
+				Weight:  sdk.NewInt(100),
+			}},
+			isPoolPrepared: false,
+			expectErr:      true,
+		},
+		{
+			name: "Adding two of the same gauge id at once should error",
+			args: []types.DistrRecord{
+				{
+					GaugeId: 1,
+					Weight:  sdk.NewInt(100),
+				},
+				{
+					GaugeId: 1,
+					Weight:  sdk.NewInt(100),
+				},
+			},
+			isPoolPrepared: true,
+			expectErr:      true,
+		},
+		{
+			name: "Adding unsort gauges at once should error",
+			args: []types.DistrRecord{
+				{
+					GaugeId: 2,
+					Weight:  sdk.NewInt(100),
+				},
+				{
+					GaugeId: 1,
+					Weight:  sdk.NewInt(100),
+				},
+			},
+			isPoolPrepared: true,
+			expectErr:      true,
+		},
+		{
+			name: "Happy case",
+			args: []types.DistrRecord{
+				{
+					GaugeId: 0,
+					Weight:  sdk.NewInt(100),
+				},
+				{
+					GaugeId: 1,
+					Weight:  sdk.NewInt(100),
+				},
+			},
+			isPoolPrepared: true,
+			expectErr:      false,
+		},
+	}
 
-	return gaugeId
+	for _, test := range tests {
+		suite.Run(test.name, func() {
+			suite.SetupTest()
+			keeper := suite.App.PoolIncentivesKeeper
+
+			if test.isPoolPrepared {
+				suite.PrepareBalancerPool()
+			}
+
+			err := keeper.UpdateDistrRecords(suite.Ctx, test.args...)
+			if test.expectErr {
+				suite.Require().Error(err)
+			} else {
+				suite.Require().NoError(err)
+
+				distrInfo := keeper.GetDistrInfo(suite.Ctx)
+				suite.Require().Equal(len(test.args), len(distrInfo.Records))
+				suite.Require().Equal(sdk.NewInt(100), distrInfo.Records[0].Weight)
+				suite.Require().Equal(sdk.NewInt(100), distrInfo.Records[1].Weight)
+				suite.Require().Equal(sdk.NewInt(200), distrInfo.TotalWeight)
```

Comment:
nit: do you want to try adding test cases for having different weights to check if the total weight is getting updated correctly?

Reply:
And then we can have expectedTotalWeight in the test case field